airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_40_17_001__AddAuditLog.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.instance.configs.migrations;
+
+import io.airbyte.commons.resources.MoreResources;
+import java.io.IOException;
+import java.util.Set;
+import java.util.UUID;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class V0_40_17_001__AddAuditLog extends BaseJavaMigration {
+
+  private static final UUID PLACEHOLDER_CREATED_BY = UUID.fromString("00000000-0000-0000-0000-000000000000");
+  private static final Field<UUID> DEFAULT_CREATED_BY_COLUMN =
+      DSL.field("updated_by", SQLDataType.UUID.nullable(false).defaultValue(PLACEHOLDER_CREATED_BY));
+
+  private static final Set<String> TABLE_TO_TRACK = Set.of(
+      "actor_definition",
+      "actor",
+      "actor_oauth_parameter",
+      "connection",
+      "connection_operation",
+      "operation",
+      "stream_reset",
+      "workspace",
+      "workspace_service_account"
+  // "state" todo (cgardens) should we do this one too?
```

Comment:
@gosusnp or @jdpgrailsdev do you have thoughts on whether we want to track state with this method? i think this is one case where it could actually cause performance issues. i'm inclined _not_ to do it.

Reply:
So far, there is no manual operation on this table and I don't think we want to have users fiddling with the state through an API so it doesn't seem as critical at this moment, however, we should remember to add it if we were to expose this feature.
What kind of performance issue are you concerned about? the extra insert from the trigger?